Elastic

vinid opened the pull request:
elasticsearch integration

vinid committed:
introduced: (i) model to represent papars in elastic and (ii) script to generate the db in elastic containing the paper

vinid committed:
- integrated elasticsearch search in the frontend
- commented out whoosh
- created a new package to handle non flask specific stuff

vinid committed:
- multi keyword search in title with elastic

vinid committed:
- multi field query

beduffy commented on app.py:
```diff
@@ -181,20 +182,28 @@ def search_papers():
+
+    return jsonify(data)
+
+#@app.route("/search-papers")
```
You can delete the below @vinid

beduffy replied:
remove the whoosh comments too

beduffy replied:
even from setup.py or requirements.txt too

beduffy commented on app.py:
```diff
@@ -377,7 +386,7 @@ def download_model(key, output_path):
 
 # Open previously created index at directory
-whoosh_ix = index.open_dir("whoosh_indexdir")
+#whoosh_ix = index.open_dir("whoosh_indexdir")
```
yeah remove this

beduffy commented on scripts/create_elastic_db.py:
```diff
@@ -0,0 +1,78 @@
+# lets load the existing database to memory
+try:
+    print()
+    db = pickle.load(open('/home/vinid/Scaricati/db_jan_13th.p', 'rb'))
```
maybe an argparse param here would be a good idea

vinid committed:
- added some comments
- create db script is now argparsed
- removed whoosh
- changed some print to logging function
